optaplanner-core/src/main/java/org/optaplanner/core/api/solver/SolverJob.java

```diff
@@ -50,8 +51,14 @@
     // TODO Future features
     //    void reloadProblem(Function<? super ProblemId_, Solution_> problemFinder);
 
-    // TODO Future features
-    //    void addProblemFactChange(ProblemFactChange<Solution_> problemFactChange);
+    /**
+     * Schedules a {@link ProblemChange} to be processed by the underlying {@link Solver} and returns immediately.
+     *
+     * @param problemChange never null
+     * @throws IllegalStateException if the underlying {@link Solver} is not in the {@link SolverStatus#SOLVING_ACTIVE}
+     *         state
```

Comment:
This creates an interesting question:
If I do solverManager.solve(dataset5) with parallelThreadCount 4 and already 4 datasets submitted,
that new dataset5 won't start solving until one of those 4 first datasets end.
If I have a ProblemChange for dataset 5 before it starts solving, I don't want to loose that change? Or maybe I can, as I 'll once dataset 5 solving starts, it starts by loading it from the database.

Damn, I smell a meeting.

Reply:
Result: in a meeting, we agreed to accept the problem change even in the SOLVING_SCHEDULED state, as at that point, the solver has been already created.